packages/devtools_app/lib/src/vm_service_wrapper.dart

```diff
@@ -766,6 +766,36 @@ class VmServiceWrapper implements VmService {
     ).isSupported(supportedVersion: supportedVersion);
   }
 
+  /// Retrieves the full string value of a [stringRef].
+  ///
+  /// The string value stored with the [stringRef] is returned unless the value
+  /// is truncated in which an extra getObject call is issued to return the
+  /// value. If the [stringRef] has expired so the full string is unavailable,
+  /// [onUnavailable] is called to return how the truncated value should be
+  /// displayed. If [onUnavailable] is not specified, an exception is thrown
+  /// if the full value cannot be retrieved.
+  Future<String> retrieveFullStringValue(
+    String isolateId,
+    InstanceRef stringRef, {
+    String onUnavailable(String truncatedValue),
+  }) async {
+    if (stringRef == null) return null;
+    if (stringRef.valueAsStringIsTruncated != true)
+      return stringRef.valueAsString;
+
+    final dynamic result = await getObject(isolateId, stringRef.id,
```

Comment:
do you have to specify dynamic here for type checking to work below?

Reply:
this method is just being moved to a central location. 
however I agree with the cleanup. removed the dynamic and used type inference.